tycho-core/src/main/java/org/eclipse/tycho/core/osgitools/DefaultArtifactDescriptor.java

```diff
@@ -68,10 +68,13 @@ public File getLocation(boolean fetch) {
         if (projectLocation != null) {
             return projectLocation;
         }
-        if (fetch && locationSupplier != null && (location == null || !location.exists())) {
+        if (fetch && locationSupplier != null && location == null) {
```

Comment:
The check was there to make sure the file was not deleted in the meantime (e.g. outside of control of the descriptor), this won't be detected anymore.

Reply:
> Also, if this is related to the "file is corrupt, trigger download again" functionality, then I think we should do something there were the download is re-triggered

The problem is that it is not that easy. We know that the file is corrupt (and can delete it) but there is no easy way to know here it was downloaded from.